### DISCUSSION TITLE
Keybind case sensitivity, capital letters, and capslock oddities

### DISCUSSION THREAD

**Discussion by alaestor** [mpv-player/mpv#15487] [General]
I haven't looked into how input is being handled, and I haven't tried this out on anything but my x11 system... I'm sure there's a fine reason, but this kind of hurts my brain.

```
shift+b print-text "shift+b"
shift+B print-text "shift+B"
b print-text "b"
B print-text "B"
```
results in... (key -> output)
```
b -> b
shift+b -> B
(capslock) b -> B
```
so capital `B` and indistinguishable from `shift+B`... What if we change the ordering?
```
b print-text "b"
B print-text "B"
shift+b print-text "shift+b"
shift+B print-text "shift+B"
```
results in...
```
b -> b
shift+b -> shift+B
(capslock) b -> shift+B
```
Okay, so capslock acts as a shift modifier. Kinda makes sense, I guess... But that's for *character keys*. What about non-character keys?
```
RIGHT print-text "right"
SHIFT+RIGHT print-text "shift+right"
```
results in...
```
right -> right
shift+right -> shift+right
(capslock) right -> right
(capslock) shift+right -> shift+right
```

So capslock acts as a shift modifier for characters, but not for non-characters.

On top of that, `shift+b` is synonymous with `shift+B`, but `ctrl+b` is distinct from `ctrl+B`, the latter implies either capslock be on or `ctrl+shift+b` be pressed...  `shift+b` can't be distinguished from `shift+B`, even if you send a lower-case `b` while holding `shift` (inverting case with capslock on)

I don't really know why I'm sharing this, but I felt like it had to be mentioned somewhere in case anyone else gets confused by this behavior while troubleshooting their `input.conf`. I had always assumed that bindings were case-insensitive and that modifiers were all that was relevant; but clearly that's not the case, which is fine, but the rules are a bit unintuitive...

Normally I'd like to see this sort of thing be simplified into a more coherent model, but I'm sure that would be a bad idea since there's bound to be a lot of [spacebar heating](https://xkcd.com/1172/)

**Comment by na-na-hi** [comment]
This is already documented in https://mpv.io/manual/master/#key-names.

I have added some clarification of this behavior to documentation in https://github.com/mpv-player/mpv/pull/15516.

**Reply by alaestor** [reply]
Thanks for the explanation & improved documentation

**Comment by brian6932** [comment]
I'm long aware of this, and due to that, I'd like some option to display a caps lock warning of sorts if possible. Kind of how the windows log-in screen, and monketype.com do. I know I can `show-text`, but it should be persistent.
